### ISSUE TITLE
Providing invalid date results in an infinite loop

### ISSUE THREAD

**Issue by undsoft** [Hexagon/croner#65]
```js
const cron2 = new Cron('0 * * * *', {
  legacyMode: true,
});
cron2.next(new Date('pizza'));
```

I think I also had an infinite loop with a valid date, but I'm still working on a concise way to reproduce it.

**Comment by Hexagon** [comment]
Solution in https://github.com/Hexagon/croner/commit/1e422848176ab2ec3362bd761c1daeabebc6c746

**Comment by Hexagon** [comment]
Fixed in 4.2.3
